src/pd/async/util.rs

```diff
@@ -171,3 +185,153 @@ impl<Req, Resp, F> Request<Req, Resp, F>
             .boxed()
     }
 }
+
+/// Do a request in synchronized fashion.
+pub fn sync_request<F, R>(client: &LeaderClient, f: F, retry: usize) -> Result<R>
```

Comment:
I prefer define `fn sync_request_with_retries` and let sync_request call this function with LEADER_CHANGE_RETRY internally.